vtrace/platforms/gdbstub.py

```diff
@@ -846,24 +932,63 @@ def getMemoryMaps(self, pgsize=4096):
                         # of the next
                         off += pgsize
                         data = self.gdbReadMem(off + qoff - 5, 10)
-                        print("DEBUG: %x:  %r" % (off+qoff-5, data))
+                        logger.debug('%#x-%#x: %r', off+qoff-5, off+qoff+5, data)
 
                 except Exception as e:
-                    print("DEBUG: %r" % e)
+                    logger.exception('failed to read %#x-%#x from gdb', off+qoff-5, off+qoff+5)
 
                 maps.append((qoff, off, 7, b'offmap-%s' % offname))
                 # TODO: scour large amount of address space looking for maps?
 
         return maps
 
-            
-    def gdbGetFeatureFile(self, fname=b'target.xml'):
-        return self.qXfer(b'features', name=fname)
-
+    def gdbGetXMLFile(self, fname=b'target.xml'):
+        if isinstance(fname, str):
+            fname = fname.encode()
+        if fname not in self._xml:
+            data = self.qXfer(b'features', name=fname)
+            if not data:
+                raise Exception('Error occurred while retrieving feature file %s' % fname)
+
+            parser = etree.XMLParser(recover=True, load_dtd=True)
+            parser.resolvers.add(GdbDTDResolver())
+            self._xml[fname] = etree.parse(StringIO(data.decode()), parser)
+        return self._xml[fname]
```

Comment:
in case this object is reused, _xml should be cleared out, perhaps in `gdbAttach()`.  i realize it may be cleaner to create a new GdbClientStub each time, but that may not always be the best option,

Reply:
oops, good call, I'll clear the cached XML information in `gdbAttach()`